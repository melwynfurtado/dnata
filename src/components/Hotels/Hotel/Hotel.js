import React from 'react'
import PropTypes from 'prop-types'
import Stars from './Stars'
import './Hotel.scss'

const Hotel = ({ hotel }) => (
  <div 
    key={hotel.EstablishmentId} 
    className="hotel card mb-3"
  >
    <div className="card-body">
      <div className="card-text">
        <img 
          className="rounded float-md-left float-sm-left mr-3" 
          src={hotel.ThumbnailUrl} 
          alt={hotel.Name} 
        />
        <div className="float-md-left float-sm-left">
          <h5 className="card-title">{ hotel.Name }</h5>
          <Stars stars={hotel.Stars} />
          <div>
            { hotel.Location }&nbsp;
            (<span className="fa fa-map-marker"></span>&nbsp;
            { Math.round(hotel.Distance * 100)/100 } miles away)
          </div>            
        </div>
        <div className="float-md-right text-md-right float-sm-left text-sm-left">
          <div>{ hotel.UserRatingTitle } - { hotel.UserRating } / 10</div>
          <div className="mb-3 text-muted">{ hotel.UserRatingCount } reviews</div>
          <div className="font-weight-bold font-italic">Â£{ hotel.MinCost } (per night)</div>
        </div>
      </div>
    </div>
  </div>
)

Hotel.propTypes = {
  hotel: PropTypes.shape({
    EstablishmentId: PropTypes.number.isRequired,
    Name: PropTypes.string,
    Distance: PropTypes.number,
    EstablishmentType: PropTypes.string,
    Location: PropTypes.string,
    MinCost: PropTypes.number,
    Stars: PropTypes.number,
    UserRating: PropTypes.number,
    UserRatingTitle: PropTypes.string,
    UserRatingCount: PropTypes.number,
    ImageUrl: PropTypes.string,
    ThumbnailUrl: PropTypes.string,
  })
}

export default Hotel