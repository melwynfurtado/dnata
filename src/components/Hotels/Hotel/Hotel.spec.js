import React from 'react'
import renderer from 'react-test-renderer'
import Hotel from './Hotel'

describe('Hotel Component', () => {
  let Component

  const hotel = {
    "Distance": 3.4872358588140977,
    "EstablishmentId": 2182944,
    "EstablishmentType": "Hotel",
    "Location": "Paris",
    "MinCost": 844,
    "Name": "Aberotel Montparnasse",
    "Stars": 3,
    "UserRating": 8,
    "UserRatingTitle": "Great",
    "UserRatingCount": 1,
    "ImageUrl": "https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_400_400.jpg",
    "ThumbnailUrl": "https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_80_80.jpg"
  }

  beforeEach(() => {
    Component = <Hotel hotel={hotel} />
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })
})