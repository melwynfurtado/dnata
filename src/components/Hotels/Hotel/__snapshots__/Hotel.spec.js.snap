// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel Component renders correctly 1`] = `
<div
  className="hotel card mb-3"
>
  <div
    className="card-body"
  >
    <div
      className="card-text"
    >
      <img
        alt="Aberotel Montparnasse"
        className="rounded float-md-left float-sm-left mr-3"
        src="https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_80_80.jpg"
      />
      <div
        className="float-md-left float-sm-left"
      >
        <h5
          className="card-title"
        >
          Aberotel Montparnasse
        </h5>
        <div
          className="stars"
        >
          <span
            className="fa fa-star checked"
          />
          <span
            className="fa fa-star checked"
          />
          <span
            className="fa fa-star checked"
          />
          <span
            className="fa fa-star false"
          />
          <span
            className="fa fa-star false"
          />
        </div>
        <div>
          Paris
            (
          <span
            className="fa fa-map-marker"
          />
           
          3.49
           miles away)
        </div>
      </div>
      <div
        className="float-md-right text-md-right float-sm-left text-sm-left"
      >
        <div>
          Great
           - 
          8
           / 10
        </div>
        <div
          className="mb-3 text-muted"
        >
          1
           reviews
        </div>
        <div
          className="font-weight-bold font-italic"
        >
          £
          844
           (per night)
        </div>
      </div>
    </div>
  </div>
</div>
`;
